firedrake/mesh.py

```diff
@@ -1294,6 +1359,35 @@ def init(self):
         if hasattr(self, '_callback'):
             self._callback(self)
 
+    def partial_init(self, topology):
```

Comment:
I think `init_topology` is a more obvious name given what it does. There's also no underscore in front of the name -- is it supposed to be part of the public API?

Reply:
Thanks, Dan. Maybe we should rename, but this method does slightly more than just setting topology, so I'm not sure what we should call it.
This method is actually called from `Mesh` function, so I think we should leave it public?

Reply:
Is it something that a user would expect to call by hand? If not, can probably be private

Reply:
I see. I'll make it private. Thanks, both.

I'm still trying to pick a name: `_init_topology`, `_partial_init`, `_init_without_coordinates`, or something else?

Reply:
Thanks. Fixed